sql/log_event.cc

```diff
@@ -93,6 +93,11 @@ TYPELIB binlog_checksum_typelib=
 
 #define FLAGSTR(V,F) ((V)&(F)?#F" ":"")
 
+#ifndef EMBEDDED_LIBRARY
+// Uses the THD to update the global stats by user name and client IP
+void update_global_user_stats(THD* thd, bool create_user, time_t now);
```

Comment:
The existing patch uses this "declarations everywhere" practice instead of a header file, this has to end - please use sql_connect.h (? not sure if actually the best place) and replace this and other existing declarations with an include